File: examples/src/bin/create_account.rs
use objectscale_client::client::ManagementClient;
use objectscale_client::iam::{AccountBuilder, Tag};

fn main() {
    let endpoint = "https://10.225.108.186:443";
    let username = "root";
    let password = "Password123!";
    let insecure = true;

    let mut client = ManagementClient::new(endpoint, username, password, insecure);
    let account = AccountBuilder::default()
        .alias("test")
        .encryption_enabled(true)
        .description("test")
        .tags(vec![Tag {
            key: "key1".to_string(),
            value: "value1".to_string(),
        }])
        .build()
        .expect("account");
    let account = client.create_account(account).expect("create account");
    println!("Created account: {:?}", account);
}

File: python/examples/create_account.py
import objectscale_client

def main():
    endpoint = "https://10.225.108.189:443"
    username = "root"
    password = "Password123@"
    insecure = True

    alias = "test"
    account = objectscale_client.iam.Account()
    account.alias = alias

    try:
        client = objectscale_client.client.ManagementClient(endpoint, username, password, insecure)
        new_account = client.create_account(account)
        print("Created account:", new_account)
    except Exception as e:
        print("Failed to create account:", e)


if __name__ == '__main__':
    main()
